src/ui/public/agg_types/metrics/lib/parent_pipeline_agg_controller.js

```diff
@@ -0,0 +1,63 @@
+const aggFilter = ['!top_hits', '!percentiles', '!percentile_ranks', '!median', '!std_dev'];
+
+const parentPipelineAggController = function ($scope) {
+
+  $scope.safeMakeLabel = function (agg) {
+    try {
+      return agg.makeLabel();
+    } catch (e) {
+      return '- agg not valid -';
+    }
+  };
+
+  $scope.$watch('responseValueAggs', updateOrderAgg);
+  $scope.$watch('agg.params.metricAgg', updateOrderAgg);
+
+  $scope.$on('$destroy', function () {
+    if ($scope.aggForm && $scope.aggForm.agg) {
+      $scope.aggForm.agg.$setValidity('bucket', true);
+    }
+  });
+
+  // Returns true if the agg is not compatible with the terms bucket
+  $scope.rejectAgg = function (agg) {
```

Comment:
Few suggestions with this function:
 - I think the name should be something like get* or is*, otherwise it sounds like the function is rejecting the agg, not checking whether it should be rejected. Maybe `isAggInvalid`?
 - Looks like the aggFilter array is only used in this function.  Can move inside and maybe rename invalidAggs?  And then I would get rid of the '!' in front of all the names.  Why is it there if it's just ignored?  Then you can simplify the logic too:
```
const invalidAggs = [...];
return Boolean(invalidAggs.find(agg.type.name);
```
 - I would update the comment because I don't think this is specific to the terms bucket.  Actually, if you rename the function to isAggInvalid, I don't even think it needs a comment as the function name will be self explanatory.